src/lib/device-state.ts

```diff
@@ -97,6 +97,33 @@ export const serviceInstallFromImage = (
 export const formatImageLocation = (imageLocation: string) =>
 	imageLocation.toLowerCase();
 
+export type DeviceConfigHook = (
+	configVars: Dictionary<string>,
+	osVersion: string,
+) => void;
+
+export interface HookRegistry {
+	deviceConfig: DeviceConfigHook[];
+}
+
+const STATE_ENDPOINT_HOOKS: HookRegistry = {
+	deviceConfig: [],
+};
+
+export function registerHook<
+	Key extends keyof HookRegistry,
+	Fn extends typeof STATE_ENDPOINT_HOOKS[Key][number]
```

Comment:
Same here, this isn't really a generic and should instead be `fn: typeof STATE_ENDPOINT_HOOKS[Key][number]`